### PR TITLE
Update inference.md

### PR DESCRIPTION
**Pull request by sekyondaMeta**
Minor update to inference doc to add vLLM_nference.py and TGI examples.

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -17,3 +17,11 @@ Here we discuss frequently asked questions that may occur and we found useful al
 4. Can I add custom datasets?
 
     Yes, you can find more information on how to do that [here](Dataset.md).
+
+5. What are the hardware SKU requirements for deploying these models?
+
+    Hardware requirements vary based on latency, throughput and cost constraints. For good latency, the models were split across multiple GPUs with tensor parallelism in a machine with NVIDIA A100s or H100s. But TPUs, other types of GPUs, or even commodity hardware can also be used to deploy these models (e.g. https://github.com/ggerganov/llama.cpp).
```
Comment (chauhang) [inline docs/FAQ.md:23]: We should add Inference optimized SKUs like Nvidia A10g, T4, L4. 

For CPU side will be good to link the Intel blog for [Accelerating Llama 2 on Intel hardware / software](https://www.intel.com/content/www/us/en/developer/articles/news/llama2.html)

File: docs/FAQ.md [hunk 2]
```diff
@@ -17,3 +17,11 @@ Here we discuss frequently asked questions that may occur and we found useful al
...
+
+6. What are the hardware SKU requirements for fine-tuning Llama pre-trained models?
+
+    Fine-tuning requirements vary based on amount of data, time to complete fine-tuning and cost constraints. To fine-tune these models we have generally used multiple NVIDIA A100 machines with data parallelism across nodes and a mix of data and tensor parallelism intra node. But using a single machine, or other GPU types are definitely possible (e.g. alpaca models are trained on a single RTX4090: https://github.com/tloen/alpaca-lora).
```
Comment (chauhang) [inline docs/FAQ.md:27]: Please add Nvidia A10g, H100s to the list

File: inference/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Inference
+
+For inference we have provided an [inference script](inference.py). Depending on the type of finetuning performed during training the [inference script](inference.py) takes different arguments.
```
Comment (chauhang) [inline inference/README.md:3]: Please add the chat completion script as well for the Llama-2-chat models

**Review by chauhang — CHANGES_REQUESTED** [review]
Thanks @sekyondaMeta for getting this updated. Left few comments

File: inference/README.md (hunk 3 above)
Comment (HamidShojanazeri) [reply inference/README.md:3]: Suggest to reformat this README to kind of cover

We support three methods for inference so far

1- inference.py script provides support for HuggingFace accelerate and PEFT fine-tuned models. For more info refer to doc.

2- vLLM that uses the paged attention idea for low latency. For more info refer to doc

3- HuggingFace Text generation, For more info refer to doc

Comment (HamidShojanazeri) [reply inference/README.md:3]: -- --model_name and other args might be better to discuss in inference.py related doc.

**Comment by sekyondaMeta** [comment]
> Thanks @sekyondaMeta for getting this updated. Left few comments

Thanks for the feedback @chauhang and @HamidShojanazeri will make the suggested updates

**Comment by sekyondaMeta** [comment]
> Thanks @sekyondaMeta for getting this updated. Left few comments

@chauhang @HamidShojanazeri I have updated the PR with your feedback. Let me know your thoughts.

**Review by chauhang — APPROVED** [review]

**Pull request merged**